### PR TITLE
Record rollouts in logs, update num examples, separate train/valid in wandb

### PR DESCRIPTION
**Pull request by jacklanchantin**
**What does this PR do? Please describe:**
A summary of the change or the issue that is fixed.

Adds three things:
1.  Logging rollouts during online training and validation (see below for how to turn on)
2. Adds logging `num_examples` and `total_num_examples` in metrics loggers for online_dpo and grpo.
3. Updates [src/fairseq2/metrics/recorders/_wandb.py](src/fairseq2/metrics/recorders/_wandb.py) to separate train/valid in wandb logs

add the following to config to turn on logging rollouts
```
criterion:
  config:
    _set_:
      loss_config:
        log_rollouts: True
 ```

**Does your PR introduce any breaking changes? If yes, please list them:**
List of all backwards-incompatible changes.

**Check list:**
- [x] Was the content of this PR **discussed and approved** via a GitHub issue? (no need for typos or documentation improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairseq2/blob/main/CONTRIBUTING.md)?
- [ ] Did you make sure that your **PR does only one thing** instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**?
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you **update the [CHANGELOG](https://github.com/facebookresearch/fairseq2/blob/main/CHANGELOG.md)**? (no need for typos, documentation, or minor internal changes)

### CODE REVIEW

File: src/fairseq2/recipes/lm/_online_finetune/_common.py [hunk 1]
```diff
@@ -540,3 +541,19 @@ def compute_token_level_entropy(logits: torch.Tensor, target_mask: torch.Tensor)
     entropy_per_seq = entropy_target_only.sum(dim=-1) / target_mask.sum(dim=-1)
 
     return entropy_per_seq
+
+
+def log_rollouts(prompt_batch: PromptBatch, rollouts, split_name, num_rollouts=1):
```
Comment (jacklanchantin) [inline src/fairseq2/recipes/lm/_online_finetune/_common.py:546]: I added to _common.py, but could have a function for each individual loss file?

File: src/fairseq2/recipes/lm/_online_finetune/_common.py [hunk 2]
```diff
@@ -540,3 +541,19 @@ def compute_token_level_entropy(logits: torch.Tensor, target_mask: torch.Tensor)
...
+    """
+    log the first num_rollouts rollouts for first prompt in the batch
+    """
+    if "prompt_raw" in prompt_batch.meta_info:
```
Comment (jacklanchantin) [inline src/fairseq2/recipes/lm/_online_finetune/_common.py:550]: we need to set this as default to get the raw text prompts during training

**Review by uralik — APPROVED** [review]
thanks!
